wsgidav/dc/pam_dc.py

```diff
@@ -47,8 +50,23 @@ def get_domain_realm(self, path_info, environ):
     def require_authentication(self, realm, environ):
         return True
 
+    def validate_user(self, user_name):
+        if user_name in self.deny_users:
```

Comment:
Thank you for your insightful comment! In *nix systems, as far as I know, usernames are case-sensitive. Since PAM authentication operates in this environment, maintaining case sensitivity ensures consistency with the underlying system. For these reasons, I believe we should preserve the current approach.

Reply:
ok, PAM [seems to be case sensitive](https://docs.posit.co/ide/server-pro/authenticating_users/pam_authentication.html) indeed.

Could you do a quick test to confirm on your system that a username is either passed in `user_name` as normalized form or that the login request is rejected before `_validate_user()` gets evened called.
(My concern was, that otherwise it would be virtually impossible to define *deny_users*, since we would have to list all permutations.)

Reply:
I guess a test could be to 

- try login as 'MaRk' (assuming that your account is 'mark')
- add 'mark' to the deny_list and try again

Reply:
Thank you for your clarification. Your security concern is a significant potential security risk in authentication. I have tried the mentioned security attack in my environment. As a result, if 'mark' is in the deny_list, the deny_list wouldn’t block the username if it has different cases (e.g., 'MaRk'); however, PAM would regard ‘MaRk’ as a different user in the system. Since 'MaRk' is not a valid user, PAM would still block the login attempt, even if the same password as 'mark' is used.

Environment:
Linux: 6.8.0-31-generic
Distro: Ubuntu 24.10 (Oracular Oriole)
PAM: 1.5.3-7ubuntu2
python-pam: 2.0.2

Reply:
Thanks for taking the time to double check